moveit_ros/planning/planning_scene_monitor/include/moveit/planning_scene_monitor/planning_scene_monitor.h

```diff
@@ -699,13 +787,23 @@ class LockedPlanningSceneRO
  * will use these, preventing other writers and readers from locking
  * the same PlanningScene at the same time.
  *
+ * The constructor supports setting an optional allowed timeout duration for
+ * locking the unique mutex. In case of a timeout, the constructor throws a
+ * planning_scene_monitor::TryLockTimeoutException.
+ *
  * @see LockedPlanningSceneRO */
-class LockedPlanningSceneRW : public LockedPlanningSceneRO
+class LockedPlanningSceneRW : public LockedPlanningScene
 {
 public:
-  LockedPlanningSceneRW(const PlanningSceneMonitorPtr& planning_scene_monitor)
-    : LockedPlanningSceneRO(planning_scene_monitor, false)
+  LockedPlanningSceneRW(const PlanningSceneMonitorPtr& planning_scene_monitor,
+                        std::chrono::milliseconds timeout = std::chrono::milliseconds::zero())
+    : LockedPlanningScene(planning_scene_monitor)
   {
+    if (planning_scene_monitor_)
+    {
+      ulock_ = std::make_shared<UniqueTimedLock>(
```

Comment:
Open for debate: The shared pointers here copy the old behavior. I'm thinking of not allowing this since it doesn't make a lot of sense to share write access between different entities. I'd suggest making `LockedPlanningSceneRW` non-copyable. `LockedPlanningSceneRO` should probably also not be using a shared pointer, acquiring another shared lock seems way more reasonable to me.

Reply:
moved to https://github.com/ros-planning/moveit/pull/2794#pullrequestreview-718589042